### PR TITLE
More misc fixes

### PR DESCRIPTION
**Pull request by hppeng-wynn**
Make builder not barf with classes without atree (to reduce confusion)
Skillpoint %effect changes (def/agi, ehp calc)

### CODE REVIEW

File: builder/index.html [hunk 1]
```diff
@@ -1275,20 +1275,7 @@
...
-                            <div class = "col spell-display dark-5 rounded dark-shadow py-2" id = "spell1-info" style="display: none;">Spell 2</div>
-                        </div>
-                        <div class = "col pe-0">
-                            <div class = "col spell-display spell-expand dark-5 rounded dark-shadow pt-2 border border-dark" id="spell2-infoAvg">spell3</div>
-                            <div class = "col spell-display dark-5 rounded dark-shadow py-2" id = "spell2-info" style="display: none;">Spell 3</div>
-                        </div>
-                        <div class = "col pe-0">
-                            <div class = "col spell-display spell-expand dark-5 rounded dark-shadow pt-2 border border-dark" id="spell3-infoAvg">spell4</div>
-                            <div class = "col spell-display dark-5 rounded dark-shadow py-2" id = "spell3-info" style="display: none;">Spell 4</div>
+                            <div class = "col spell-display spell-expand dark-5 rounded dark-shadow pt-2 border border-dark" id="spell0-infoAvg">Input a weapon to see abilities!</div>
```
Comment (ferricles) [inline builder/index.html:1278]: idk if this is how the system works rn but using spell0-infoAvg as an id might get confused later in the system for an actual 0th (1st in array) spell or smth

i assume that this div and its container (col pe-0) get deleted too when a weapon gets input

File: js/atree.js [hunk 2]
```diff
@@ -473,7 +475,7 @@ const atree_collect_spells = new (class extends ComputeNode {
                 case 'add_spell_prop': {
                     const { base_spell, target_part = null, cost = 0, behavior = 'merge'} = effect;
                     if (base_spell !== base_spell_id) { continue; }   // TODO: redundant? if we assume abils only affect one spell
-                    ret_spell.cost += cost;
+                    if ('cost' in ret_spell) { ret_spell.cost += cost; }
```
Comment (ferricles) [inline js/atree.js:478]: is cost decremental on upgrades? 

just a sanity check

File: js/builder_graph.js [hunk 3]
```diff
@@ -502,16 +502,19 @@ class SpellSelectNode extends ComputeNode {
  */
 function getDefenseStats(stats) {
     let defenseStats = [];
-    let def_pct = skillPointsToPercentage(stats.get('def'));
-    let agi_pct = skillPointsToPercentage(stats.get('agi'));
+    let def_pct = skillPointsToPercentage(stats.get('def')) * 0.867;    // lol
+    let agi_pct = skillPointsToPercentage(stats.get('agi')) * 0.951;    // lol2
```
Comment (ferricles) [inline js/builder_graph.js:506]: uhhhhhh where did these constants come from

should either have them as constants in code or document how they got here

File: builder/index.html (hunk 1 above)
Comment (hppeng-wynn) [reply builder/index.html:1278]: yeah, all gets deleted before anything is updated

todo: make melee display better...

File: js/atree.js (hunk 2 above)
Comment (hppeng-wynn) [reply js/atree.js:478]: on upgrades cost is input as negative if its reduced ye
~~it works on my machine~~

File: js/builder_graph.js (hunk 3 above)
Comment (hppeng-wynn) [reply js/builder_graph.js:506]: yeah actually i need to fix this lol
they're used in 2 separate places 🥴 bad code

**Pull request merged**